backtracking.go

```diff
@@ -32,7 +32,7 @@ type Backtracking struct {
 	initG    float64
 }
 
-func (b *Backtracking) Init(loc LinesearchLocation, step float64, _ *FunctionInfo) EvaluationType {
+func (b *Backtracking) Init(loc LinesearchLocation, step float64, _ *Problem) EvaluationType {
```

Comment:
I know FunctionInfo seems redundant, but it's not. A big invariant in this package is that optimization methods cannot call the functions directly. Giving them access through Problem makes it very tempting to cheat, which breaks all measurement (of things like NumEvals)